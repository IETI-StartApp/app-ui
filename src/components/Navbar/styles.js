import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles((theme) => ({
    appBar: {
        backgroundColor: '#fff',
    },
    menuButton: {
        marginRight: theme.spacing(2),
        color: '#950740'
    },
    title: {
        flexGrow: 1,
        color: '#950740'
    },
    buttonBar: {
        color: '#950740',
        marginRight: theme.spacing(2),
    },
}));