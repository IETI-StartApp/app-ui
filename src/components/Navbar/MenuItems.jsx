import React from 'react';
import List from '@material-ui/core/List';
import { ListItem, ListItemIcon, ListItemText } from '@material-ui/core/';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import SearchIcon from '@material-ui/icons/Search';
import AddIcon from '@material-ui/icons/Add';
import { useStyles } from './styles';
import BarChartIcon from '@material-ui/icons/BarChart';
import {useHistory} from "react-router";
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';
import VideoCallIcon from '@material-ui/icons/VideoCall';

export default function MenuItems( { userType } ) {
    const history = useHistory() || [];
    const classes = useStyles()
    const investorItems = [
        {text:"Mis inversiones", icon: <StarBorderIcon />, path: '/'},
        {text:"Buscar proyectos", icon: <SearchIcon />, path: '/projects'},
        {text:"Proyectos guardados", icon: <BookmarkBorderIcon />, path: '/projects'},
        {text:"Comunicarse con un asesor", icon: <VideoCallIcon />, path: '/'}
    ];
    const enterprisingItems = [
        {text:"Nuevo proyecto", icon: <AddIcon />, path: '/register-project'},
        {text:"Mis proyectos", icon: <BarChartIcon />, path: '/'},
        {text:"Buscar proyectos", icon: <SearchIcon />, path: '/projects'},
        {text:"Comunicarse con un asesor", icon: <VideoCallIcon />, path: '/'}
    ];
    const getMenuItems = () => {
        if (userType === "Investor"){
            return investorItems;
        } return enterprisingItems;
    }

    return (
        <List component='nav'>
            {
                getMenuItems().map( (item) => (
                    <ListItem button className={classes.mainColor} key={item.text} onClick={() => history.push(item.path)}>
                        <ListItemIcon className={classes.mainColor}>
                            {item.icon}
                        </ListItemIcon>
                        <ListItemText primary={item.text} />
                    </ListItem>
                ))
            }
        </List>
    );

}