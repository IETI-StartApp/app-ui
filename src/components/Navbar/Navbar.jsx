import React from 'react';
import { AppBar, Toolbar, IconButton, Button, Typography } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import { useStyles } from './styles';

export default function Navbar({ handleDrawerToggle }) {

    const classes = useStyles();

    return (
        <AppBar position="static" className={classes.appBar}>
            <Toolbar>
                <IconButton
                    edge="start"
                    className={classes.menuButton}
                    color="inherit"
                    aria-label="menu"
                    onClick={() => handleDrawerToggle()}
                >
                    <MenuIcon />
                </IconButton>
                <Typography variant="h6" className={classes.title}>
                    StartApp
                </Typography>
                <Button className={classes.buttonBar}>Opción 1</Button>
                <Button className={classes.buttonBar}>Opción 2</Button>
                <Button className={classes.buttonBar} variant="outlined" >Opción 3</Button>
            </Toolbar>
        </AppBar>
    );
}