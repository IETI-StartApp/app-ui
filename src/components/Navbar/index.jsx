import React from 'react';
import {AppBar, Button, Hidden, IconButton, Toolbar, Typography, CssBaseline } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import {useStyles} from './styles';
import {useAuth} from "../../authServices/Auth";
import {useHistory} from "react-router";
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import HomeIcon from '@material-ui/icons/Home';

export default function Navbar({handleDrawerToggle, test}) {
    const classes = useStyles();
    const {signOut} = useAuth() || {};
    const history = useHistory() || [];
    const buttons = <>
        <Button className={classes.buttonBar} onClick={() => history.push('/dashboard')} startIcon={<HomeIcon />}>Inicio
        </Button>
        <Button className={classes.buttonBar} onClick={signOut} startIcon={<ExitToAppIcon />}>Salir
        </Button>
    </>
    return (
        <div className={classes.root}>
            <CssBaseline />
            <AppBar position="fixed" className={classes.appBar}>
                <Toolbar>
                    <Hidden mdUp>
                        <IconButton
                            edge="start"
                            className={classes.menuButton}
                            color="inherit"
                            aria-label="menu"
                            onClick={() => handleDrawerToggle()}
                        >
                            <MenuIcon/>
                        </IconButton>
                    </Hidden>
                    <Typography variant="h6" className={classes.title}>
                        StartApp
                    </Typography>
                    {
                        test && buttons
                    }
                    {
                        buttons
                    }
                </Toolbar>
            </AppBar>
        </div>
    );
}
