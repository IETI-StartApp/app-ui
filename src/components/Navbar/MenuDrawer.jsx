import React from 'react';
import { Drawer } from '@material-ui/core/';
import { useStyles } from './styles';
import MenuItems from './MenuItems';
import { useAuth } from "../../authServices/Auth";
import Typography from "@material-ui/core/Typography";

export default function MenuDrawer({ open, onClose, variant }) {
    const classes = useStyles();
    const { currentUser } = useAuth() || { currentUser: { photoURL: '', displayName: '' } };

    return (
        <Drawer
            className={classes.drawer}
            classes={{
                paper: classes.drawerPaper,
            }}
            anchor="left"
            variant={variant}
            open={open}
            onClose={onClose ? onClose : null}
        >
            <div className={classes.menuHeader} style={{ display: "flex", alignItems: "center" }}>
                <img src={currentUser.photoURL} alt="" />
                <Typography style={{ paddingLeft: "20px" }}>
                    {currentUser.displayName}
                </Typography>
            </div>
            <MenuItems userType="Entrepreneur" />
        </Drawer>
    );
}
