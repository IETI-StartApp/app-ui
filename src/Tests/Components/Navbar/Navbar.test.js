import Navbar from '../../../components/Navbar/index';
import { IconButton, Button} from '@material-ui/core';
import { shallow } from 'enzyme';

describe('Testing navbar', () => {

    const handleDrawerToggle = jest.fn();

    const wrapper = shallow(<Navbar handleDrawerToggle={handleDrawerToggle}/>)

    test('should load navbar ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should call handleDrawerToggle function', () => {
        const iconButton = wrapper.find(IconButton).at(0);
        iconButton.simulate("Click");
        expect( handleDrawerToggle ).toHaveBeenCalled();
    });

    test("should load 2 buttons", () => {
        const button1 = wrapper.find(Button).at(0);
        const button2 = wrapper.find(Button).at(0);
        expect( button1 ).toBeDefined();
        expect( button2 ).toBeDefined();
    });

})