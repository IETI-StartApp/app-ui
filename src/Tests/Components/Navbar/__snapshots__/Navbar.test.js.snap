// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing navbar should load navbar  1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <Hidden
        mdUp={true}
      >
        <WithStyles(ForwardRef(IconButton))
          aria-label="menu"
          className="makeStyles-menuButton-3"
          color="inherit"
          edge="start"
          onClick={[Function]}
        >
          <Memo([object Object]) />
        </WithStyles(ForwardRef(IconButton))>
      </Hidden>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-5"
        variant="h6"
      >
        StartApp
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-buttonBar-6"
        onClick={[Function]}
        startIcon={<Memo />}
      >
        Inicio
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-buttonBar-6"
        startIcon={<Memo />}
      >
        Salir
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
